File: src/CSPAF/CSPCore/CSPRedwoodHQ/CSPMarkActionAsPassed.cs
using System;
using System.Linq;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Services;
using CloudSuitePortal.Services.RedwoodHQ;

namespace actions.CSP
{
    class CSPMarkActionAsPassed
    {
        public void run(Dictionary<string, object> Params)
        {
            var id = (string)Params["Id"];
            
            var actionManager = new CSPActionManager();
            var action = actionManager.Action(a => a.Name == id);
            action.CurrentState = CSPActionItem.State.Passed;
            actionManager.SaveAction(action);
        }
    }
}

File: src/CSPAF/CSPCore/CSPRedwoodHQ/CSPMarkActionAsPassed.java
package CSPAF.CSPCore.CSPRedwoodHQ;

import java.util.*;

import com.infor.cloudsuite.portal.core.helpers.*;
import com.infor.cloudsuite.portal.core.services.*;

class CSPMarkActionAsPassed
{
    public void run(HashMap<String, Object> params) throws Exception
    {
        String id = params.get("Id").toString();
            
        CSPActionManager actionManager = new CSPActionManager();
        CSPActionItem action = actionManager.action(actionItem -> actionItem.name.equals(id));
        action.currentState = CSPActionItem.State.Passed;
        actionManager.saveAction(action);
    }
}